File: src/main/java/com/aquasecurity/plugins/trivy/ui/FindingTreeRenderer.java
package com.aquasecurity.plugins.trivy.ui;

import com.aquasecurity.plugins.trivy.ui.treenodes.TrivyTreeNode;
import com.intellij.ui.JBColor;

import javax.swing.*;
import javax.swing.tree.DefaultTreeCellRenderer;
import java.awt.*;

class FindingTreeRenderer extends DefaultTreeCellRenderer {

    public FindingTreeRenderer() {
    }

    public Component getTreeCellRendererComponent(
            JTree tree,
            Object value,
            boolean sel,
            boolean expanded,
            boolean leaf,
            int row,
            boolean hasFocus) {

        super.getTreeCellRendererComponent(
                tree, value, sel,
                expanded, leaf, row,
                hasFocus);

        if (value instanceof TrivyTreeNode) {
            TrivyTreeNode node = (TrivyTreeNode) value;
            setIcon(node.getIcon());
            setText(node.getTitle());
            setBackgroundNonSelectionColor(JBColor.PanelBackground);
            setToolTipText(node.getTooltip());
        }
        return this;
    }

}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/FindingTreeRenderer.kt
package com.aquasecurity.plugins.trivy.ui

import com.aquasecurity.plugins.trivy.ui.treenodes.TrivyTreeNode
import java.awt.Color
import java.awt.Component
import javax.swing.JTree
import javax.swing.tree.DefaultTreeCellRenderer

internal class FindingTreeRenderer : DefaultTreeCellRenderer() {
  companion object {
    private val TransparentColor = Color(0, 0, 0, 0)
  }

  override fun getTreeCellRendererComponent(
      tree: JTree,
      value: Any,
      sel: Boolean,
      expanded: Boolean,
      leaf: Boolean,
      row: Int,
      hasFocus: Boolean,
  ): Component {
    super.getTreeCellRendererComponent(tree, value, sel, expanded, leaf, row, hasFocus)

    if (value is TrivyTreeNode) {
      val node = value
      icon = node.icon
      text = node.title
      setBackgroundNonSelectionColor(TransparentColor)
      toolTipText = node.tooltip
    }
    return this
  }
}
